eclipse-collections-api/src/main/java/org/eclipse/collections/api/stack/ImmutableStack.java

```diff
@@ -52,6 +53,20 @@
 
     ImmutableStack<T> pop(int count);
 
+    /**
+     * @since 11.1
+     */
+    default Pair<T, ImmutableStack<T>> peekAndPop() {
+        throw new UnsupportedOperationException("Not implemented yet!");
```

Comment:
Change the message from "Not implemented yet!" to "Concrete class responsible for implementation" since we won't be able to implement this without making it possible to create Pairs in the api module.